lib/src/builder/builder.dart

```diff
@@ -1,8 +1,144 @@
+      log.severe('There was an error parsing the compilation unit for file: ${buildStep.inputId}');
+      log.fine(error);
+      log.fine(stackTrace);
+      return;
+    }
+
+    if (isPart(libraryUnit)) {
+      return;
+    }
+
+    final librarySourceFile = new SourceFile.fromString(
+      source,
+      url: idToPackageUri(buildStep.inputId));
+    final libraryDeclarations = new ParsedDeclarations(
+      libraryUnit,
+      librarySourceFile,
+      log);
+    if (libraryDeclarations.hasErrors) {
+      log.severe('There was an error parsing the file declarations for file: ${buildStep.inputId}');
+      return;
+    }
+
+    final outputs = <String>[];
+    final libraryGenerator = new ImplGenerator(log, librarySourceFile)
+      ..generate(libraryDeclarations);
+    outputs.add(libraryGenerator.outputContentsBuffer.toString());
+
+    final parts = libraryUnit.directives
+      .whereType<PartDirective>()
+      // Ignore `.over_react.g.dart` parts - that's what we're generating.
+      .where((part) => !part.uri.stringValue.endsWith(outputExtension));
+
+    for (final part in parts) {
+      final partId = new AssetId.resolve(
+        part.uri.stringValue,
+        from: buildStep.inputId);
+      final partSource = await buildStep.readAsString(partId);
+      final partSourceFile = new SourceFile.fromString(
+        partSource,
+        url: idToPackageUri(partId));
+      CompilationUnit partUnit;
+      try {
+        partUnit = parseCompilationUnit(
+          partSource,
+          name: partId.path,
+          suppressErrors: false,
+          parseFunctionBodies: true);
+      } catch (error, stackTrace) {
+        log.severe('There was an error parsing the compilation unit for file: $partId');
+        log.fine(error);
+        log.fine(stackTrace);
+        return;
+      }
+      final partDeclarations = new ParsedDeclarations(partUnit, partSourceFile, log);
+      if (partDeclarations.hasErrors) {
+        log.severe('There was an error parsing the file declarations for file: ${buildStep.inputId}');
+        return;
+      }
+      final partGenerator = new ImplGenerator(log, partSourceFile)
+        ..generate(partDeclarations);
+      outputs.add(partGenerator.outputContentsBuffer.toString());
+    }
+
+    final nonEmptyOutputs = outputs
+      .map((s) => s.trim())
+      .where((s) => s.isNotEmpty);
+    if (nonEmptyOutputs.isEmpty) {
+      return;
+    }
+
+    await _writePart(buildStep, nonEmptyOutputs);
+  }
+
+  FutureOr<void> _writePart(BuildStep buildStep, Iterable<String> outputs) async {
+    final outputId = buildStep.inputId.changeExtension(outputExtension);
+    final partOf = "'${p.basename(buildStep.inputId.uri.toString())}'";
+
+    final buffer = new StringBuffer()
+      ..writeln('// GENERATED CODE - DO NOT MODIFY BY HAND')
+      ..writeln()
+      ..writeln('part of $partOf;')
+      ..writeln()
+      ..writeln(_headerLine)
+      ..writeln('// OverReactGenerator')
```

Comment:
#nit Personally, I'd advocate for not using the same header as the generator. It could lead to confusion if a consumer is examining our outputs.

Reply:
Fair enough, I was going for minimal changes to generated code but I guess that doesn't really matter since we're building to cache.